Library/OcBootManagementLib/VariableManagement.c

```diff
@@ -312,6 +314,109 @@ InternalGetBootstrapBootData (
   return OptionData;
 }
 
+EFI_STATUS
+OcGetSip (
+  UINT32 *CsrActiveConfig,
+  UINT32 *Attributes
+  )
+{
+  EFI_STATUS      Status;
+  UINTN           DataSize;
+
+  *CsrActiveConfig = 0;
+  DataSize = sizeof (*CsrActiveConfig);
+
+  Status = gRT->GetVariable (
+    L"csr-active-config",
+    &gAppleBootVariableGuid,
+    Attributes,
+    &DataSize,
+    CsrActiveConfig);
+
+  return Status;
+}
+
+EFI_STATUS
+OcSetSip (
+  UINT32 *CsrActiveConfig,
+  UINT32 Attributes
+  )
+{
+  EFI_STATUS      Status;
+
+  Status = gRT->SetVariable (
+    L"csr-active-config",
+    &gAppleBootVariableGuid,
+    Attributes,
+    CsrActiveConfig == NULL ? 0 : sizeof (*CsrActiveConfig),
+    CsrActiveConfig);
+
+  return Status;
+}
+
+BOOLEAN
+OcIsSipEnabled (
+  EFI_STATUS  FetchStatus,
+  UINT32      CsrActiveConfig
+  )
+{
+  return FetchStatus == EFI_NOT_FOUND || (CsrActiveConfig & ~CSR_ALLOW_APPLE_INTERNAL) == 0;
```

Comment:
I do not understand why Apple Internal is treated differently. It is not like many users can run HomeDiagnostic in the first place, and not many people in Apple would use OpenCore.

Reply:
I do now know what it does - I started to look, but did not get to the bottom of it. I was just matching the behaviour of current macOS on my Mac (i.e. of csrutil). We may need to do some more tests.

Reply:
csrutil should not set Apple Internal. Something is wrong with your computer. You may have AppleInternal somewhere due to a bug in macOS updates, thus the bit. Setting Apple Internal bit will also lead to broken updates without Apple Connect, so please do not do this.